cmd/opentelemetry-collector/main.go

```diff
@@ -61,14 +61,26 @@ func main() {
 	}
 
 	cmpts := defaults.Components(v)
-	var cfgFactory service.ConfigFactory
-	if getOTELConfigFile() == "" {
-		log.Println("Config file not provided, installing default Jaeger components")
-		cfgFactory = func(*viper.Viper, config.Factories) (*configmodels.Config, error) {
-			collectorOpts := &collectorApp.CollectorOptions{}
-			collectorOpts.InitFromViper(v)
-			return defaults.Config(storageType, collectorOpts.CollectorZipkinHTTPHostPort, cmpts)
+	cfgFactory := func(otelViper *viper.Viper, f config.Factories) (*configmodels.Config, error) {
+		collectorOpts := &collectorApp.CollectorOptions{}
+		collectorOpts.InitFromViper(v)
+		cfg, err := defaults.Config(storageType, collectorOpts.CollectorZipkinHTTPHostPort, cmpts)
+		if err != nil {
+			return nil, err
+		}
+
+		var otelCfg *configmodels.Config
+		if len(getOTELConfigFile()) > 0 {
+			otelCfg, err = service.FileLoaderConfigFactory(otelViper, f)
+			if err != nil {
+				return nil, err
```

Comment:
Should we log an error here? Indicating a config load failure?

Reply:
It will be logged in OTEL collector.